java-checks/src/main/java/org/sonar/java/checks/RedundantThrowsDeclarationCheck.java

```diff
@@ -236,10 +272,21 @@ private static boolean emptyBody(MethodTree methodTree) {
 
     @Override
     public void visitMethodInvocation(MethodInvocationTree tree) {
+      if (isSuperInvocation(tree)) {
+        visitedSuper = true;
+      }
       addThrownTypes(tree.symbol());
       super.visitMethodInvocation(tree);
     }
 
+    private static boolean isSuperInvocation(MethodInvocationTree tree) {
+      if (tree.methodSelect().is(Tree.Kind.IDENTIFIER)) {
```

Comment:
this is not enough. It won't cover cases of nested classes, called for instance with something such as `A.this.super()`.

```java
class A {

  class Parent {
    public Parent() throws IllegalAccessException { // Compliant
      throw new IllegalAccessException();
    }
  }

  class Child extends Parent {

    public Child(Long a) throws IllegalAccessException { // Compliant
      A.this.super();
      // no implicit call to parent constructor
      System.out.println("a:" + a);
    }
  }
}
```

However... You are lucky. It seems that we are not resolving properly this particular call. And it is resolved as `unknownSymbol`, which makes the rule say nothing.

To me, the proper way should however be to check that the method symbol name is `"<init>"` and that the owner of the super constructor is the parent class of the present constructor.

Reply:
I'm not so lucky:
```
class A {
  class Parent {
    public Parent() throws IllegalAccessException { // Compliant
      throw new IllegalAccessException();
    }
    public Parent(String a) { // Compliant
    }
  }

  class Child extends Parent {
    public Child(Long a) throws IllegalAccessException { // Noncompliant
      A.this.super("a:" + a);
    }
  }
}
```

This is a FN, since it does not detect that super is already called!
I will use your way

Reply:
I'm not so lucky:
```
class A {
  class Parent {
    public Parent() throws IllegalAccessException { // Compliant
      throw new IllegalAccessException();
    }
    public Parent(String a) { // Compliant
    }
  }

  class Child extends Parent {
    public Child(Long a) throws IllegalAccessException { // Noncompliant
      A.this.super("a:" + a);
    }
  }
}
```

This is a FN, since it does not detect that super is already called!
I will use your way